### PR TITLE
feat: Migrate drive init endpoint to V2

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: kDriveCore/Data/Cache/DriveInfosManager.swift [hunk 1]
```diff
@@ -240,17 +236,12 @@ public class DriveInfosManager {
...
     func storeDriveResponse(user: InfomaniakCore.UserProfile, driveResponse: DriveResponse) -> [Drive] {
         var driveList = [Drive]()
-        for drive in driveResponse.drives.main {
-            initDriveForRealm(drive: drive, userId: user.id, sharedWithMe: false)
-            driveList.append(drive)
-        }
-
-        for drive in driveResponse.drives.sharedWithMe {
-            initDriveForRealm(drive: drive, userId: user.id, sharedWithMe: true)
+        for drive in driveResponse.drives where drive.role != "none" {
```
Comment (adrien-coye) [inline kDriveCore/Data/Cache/DriveInfosManager.swift:239]: Do we have a const for "none" we could use here ?

File: kDriveCore/Data/Models/Drive/Drive.swift [hunk 2]
```diff
@@ -164,24 +118,36 @@ public class Drive: Object, Codable {
...
+    public var pack: DrivePack {
+        return _pack ?? DrivePack()
+    }
+
     public var isUserAdmin: Bool {
         return role == "admin"
     }
 
+    public var isDriveUser: Bool {
+        return role != "none" && role != "external"
```
Comment (adrien-coye) [inline kDriveCore/Data/Models/Drive/Drive.swift:134]: Same comment about "none" here.

**Review by adrien-coye — COMMENTED** [review]
All good for me, I have remark on a tiny detail. I will approve quickly.

File: kDriveCore/Data/Models/Drive/Drive.swift (hunk 2 above)
Comment (PhilippeWeidmann) [reply kDriveCore/Data/Models/Drive/Drive.swift:134]: Ideally `role` should be placed in an enum but as I haven't consulted the back-end on the possible values it is like this for now.

Comment (adrien-coye) [reply kDriveCore/Data/Models/Drive/Drive.swift:134]: ok perfect, I approve the PR then. Thanks.

**Review by adrien-coye — APPROVED** [review]
All good for a merge. You can do it at your discression.

**Comment by adrien-coye** [comment]
related to https://github.com/Infomaniak/ios-kDrive/issues/555

**Pull request merged**